WordPress/Jetpack/Classes/System/JetpackWindowManager.swift

```diff
@@ -43,20 +37,35 @@ class JetpackWindowManager: WindowManager {
                 NotificationCenter.default.post(name: .WPAccountDefaultWordPressComAccountChanged, object: self)
                 self.showMigrationUIIfNeeded(blog)
                 self.sendMigrationEmail()
-            case .failure:
-                failureCompletion?()
+            case .failure(let error):
+                self.handleMigrationFailure(error)
             }
         }
     }
+}
+
+// MARK: - Private Helpers
+
+private extension JetpackWindowManager {
+
+    var shouldShowMigrationUI: Bool {
+        return FeatureFlag.contentMigration.enabled && AccountHelper.isLoggedIn
+    }
 
-    private func sendMigrationEmail() {
+    /// Checks whether the WordPress app supports the custom scheme meant to disable notifications.
+    /// Since the scheme is added in 21.3, we can be pretty sure that the WP version also supports migration.
+    var isCompatibleWordPressAppPresent: Bool {
+        JetpackNotificationMigrationService.shared.isMigrationSupported
```

Comment:
I'd thought about using that notifications scheme for this detection [when implementing this](https://github.com/wordpress-mobile/WordPress-iOS/blob/5ab373980120459dd3e3bc5e166560f50512f6a3/WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack%20Migration/Common/MigrationAppDetection.swift#L17), but didn't know how tightly coupled they were and if we'd need to make a "V2" of the migration procedure down the road.

Reply:
Oh, good point. I'll use the `MigrationAppDetection`instead!